google/cloud/pubsub/internal/tracing_batch_callback_test.cc

```diff
@@ -104,6 +105,40 @@ TEST(TracingBatchCallback, StartAndEndMessage) {
                             SpanHasEvents(EventNamed("gl-cpp.ack_end")))));
 }
 
+TEST(TracingBatchCallback, VerifySpanIsSetInUserCallback) {
+  auto span_catcher = InstallSpanCatcher();
+  auto mock = std::make_shared<pubsub_testing::MockBatchCallback>();
+  EXPECT_CALL(*mock, callback).Times(1);
+  // Since there is no default constructor, we need to create a fake before we
+  // get the actual value.
+  MessageCallback::MessageAndHandler actual_message_and_handler{
+      pubsub::MessageBuilder().Build(),
+      std::make_unique<pubsub_testing::MockExactlyOnceAckHandlerImpl>(),
+      "invalid-ack-id", Span{}};
+  EXPECT_CALL(*mock, user_callback)
+      .WillOnce([&](MessageCallback::MessageAndHandler arg0) {
+        actual_message_and_handler = std::move(arg0);
```

Comment:
You should set the expectations in here instead of trying to extract the `MessageAndHandler`.

Also `arg0` is not an informative variable name. `m` would be fine.

Reply:
Done